netty/src/test/java/io/grpc/netty/NettyClientTransportTest.java

```diff
@@ -152,6 +154,26 @@ public void headersShouldOverrideDefaultUserAgent() throws Exception {
             receivedHeaders.get(USER_AGENT_KEY));
   }
 
+  @Test
+  public void maxMessageSizeShouldBeEnforced() throws Throwable {
+    startServer();
+    // Allow the response payloads of up to 1 byte.
+    NettyClientTransport transport = newTransport(newNegotiator(), 1);
+    transport.start(clientTransportListener);
+
+    try {
+      // Send a single RPC and wait for the response.
+      new Rpc(transport).halfClose().waitForResponse();
+    } catch (ExecutionException e) {
+      // Unpack the exception.
+      Throwable cause = e.getCause();
```

Comment:
Optional:  if you are going to dive into the cause, you might as well walk the causal chain.  Throwables has a method for it I think.